### PR TITLE
Change Nullability annotation on `ISelectionService.PrimarySelection`

### PR DESCRIPTION
**Pull request by elachlan**
The docs do not say its nullable, but winforms implementation has it returning a null value when there isn't anything selected.

https://learn.microsoft.com/en-us/dotnet/api/system.componentmodel.design.iselectionservice.primaryselection?view=net-6.0

Related to ongoing work to null annotate the winforms codebase.

https://github.com/dotnet/winforms/issues/8342

### CODE REVIEW

**Comment by elachlan** [comment]
CC: @lonitra

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/ISelectionService.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@ public interface ISelectionService
         /// <summary>
         /// Gets the object that is currently the primary selection.
         /// </summary>
-        object PrimarySelection { get; }
+        object? PrimarySelection { get; }
```
Comment (stephentoub) [inline src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/ISelectionService.cs:16]: https://github.com/dotnet/runtime/blob/339a204e35e7c6ed66c0a4ee86543dd282fa46b4/src/libraries/System.ComponentModel.TypeConverter/ref/System.ComponentModel.TypeConverter.cs#L1946 needs to be updated as well. Thanks.

Comment (elachlan) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/ISelectionService.cs:16]: done.

**Review by stephentoub — APPROVED** [review]
Thanks.

**Pull request merged**